features/filesystem/bd/ExhaustibleBlockDevice.cpp

```diff
@@ -79,17 +78,26 @@ int ExhaustibleBlockDevice::erase(bd_addr_t addr, bd_size_t size)
 {
     MBED_ASSERT(is_valid_erase(addr, size));
 
-    // use an erase cycle
-    if (_erase_array[addr / get_erase_size()] > 0) {
-        _erase_array[addr / get_erase_size()] -= 1;
-    }
+    bd_size_t eu_size = get_erase_size();
+    while (size) {
+        // use an erase cycle
+        if (_erase_array[addr / eu_size] > 0) {
+            _erase_array[addr / eu_size] -= 1;
+        }
 
-    if (_erase_array[addr / get_erase_size()] == 0) {
-        // TODO possibly something more destructive here
-        return 0;
+        if (_erase_array[addr / eu_size] == 0) {
+            return 0;
```

Comment:
if the erase expands over several sectors and only one of them was exhausted do you stop in the middle of the erase while and not erase the other sectors? Also, is there no error indication for cycles exhausted?

Reply:
Regarding the loop exit: Good point, will fix.
Regarding the error indication: This was originally included in the PR (see the strikethrough test in the description) and then reverted it. Reason was the fact that flash components don't give you an indication for it (you won't get an error, just read the wrong data). In addition, This broke the LittleFS resilience test.